src/sentry/tasks/data_export.py

```diff
@@ -13,27 +13,38 @@
 SNUBA_MAX_RESULTS = 1000
 
 
+class DataExportError(Exception):
+    pass
+
+
 @instrumented_task(name="sentry.tasks.data_export.assemble_download", queue="data_export")
 def assemble_download(data_export):
     # Create a temporary file
-    with tempfile.TemporaryFile() as tf:
-        # Process the query based on its type
-        if data_export.query_type == ExportQueryType.DISCOVER_V2:
-            process_discover_v2(data_export, tf)
-            return
-        elif data_export.query_type == ExportQueryType.BILLING_REPORT:
-            process_billing_report(data_export, tf)
-            return
-        elif data_export.query_type == ExportQueryType.ISSUE_BY_TAG:
-            file_name = process_issue_by_tag(data_export, tf)
-        # Create a new File object and attach it to the ExportedData
-        tf.seek(0)
-        with transaction.atomic():
-            file = File.objects.create(
-                name=file_name, type="export.csv", headers={"Content-Type": "text/csv"}
-            )
-            file.putfile(tf)
-            data_export.finalize_upload(file=file)
+    try:
+        with tempfile.TemporaryFile() as tf:
+            # Process the query based on its type
+            if data_export.query_type == ExportQueryType.DISCOVER_V2:
+                process_discover_v2(data_export, tf)
+                return
+            elif data_export.query_type == ExportQueryType.BILLING_REPORT:
+                process_billing_report(data_export, tf)
+                return
+            elif data_export.query_type == ExportQueryType.ISSUE_BY_TAG:
+                file_name = process_issue_by_tag(data_export, tf)
+            # Create a new File object and attach it to the ExportedData
+            tf.seek(0)
+            try:
+                with transaction.atomic():
+                    file = File.objects.create(
+                        name=file_name, type="export.csv", headers={"Content-Type": "text/csv"}
+                    )
+                    file.putfile(tf)
+                    data_export.finalize_upload(file=file)
+            except IntegrityError:
+                raise DataExportError("Failed to save the assembled file")
+    except DataExportError as err:
```

Comment:
@leeandher This is only going to catch your raised errors. You probably should have a catch block for any type of error just in case.

Reply:
Done below -> https://github.com/getsentry/sentry/pull/17267/files#diff-2775b55b64d1e5c5e480b82daaf33294R48-R50